custom_components/unifiprotect/views.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import collections
+from http import HTTPStatus
+import logging
+from typing import Any
+
+from aiohttp import web
+from homeassistant.components.http import KEY_AUTHENTICATED, HomeAssistantView
+from homeassistant.core import HomeAssistant
+from pyunifiprotect.api import ProtectApiClient
+from pyunifiprotect.exceptions import NvrError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ThumbnailProxyView(HomeAssistantView):
+    """View to proxy event thumbnails from UniFi Protect"""
+
+    requires_auth = False
+    url = "/api/ufp/thumbnail/{event_id}"
+    name = "api:ufp_thumbnail"
+
+    def __init__(self, hass: HomeAssistant):
+        self.hass = hass
+        self.data = hass.data[DOMAIN]
+
+    def _get_access_tokens(
+        self, entity_id: str
+    ) -> tuple[collections.deque, ProtectApiClient] | None:
+
+        for entry in self.data.values():
+            if entity_id in entry.protect_data.access_tokens:
+                return entry.protect_data.access_tokens[entity_id], entry.protect
```

Comment:
Linear search, but probably doesn't matter since they are unlikely to have many of them

Reply:
Yeah, that is just if there are multiple Protect Entries. Likely not going to be more than a 1 or 2.